scripts/release.js

```diff
@@ -20,29 +22,57 @@ const packageJson = require('../package.json');
 
 const {showChangelog, getChangelog, updateChangelog} = require('./lib/conventionalChangelog.js');
 
-shell.echo(`Algoliasearch-Helper release script`);
+const {canary: isCanary} = argv;
+
+if (isCanary) {
+  releaseCanaryVersion();
+} else {
+  releaseStableVersion();
+}
 
-checkEnvironment();
-mergeDevIntoMaster();
-showChangelog(shell);
-promptVersion(packageJson.version, (version) => {
-  bumpVersion(version, () => {
+function releaseCanaryVersion() {
+  shell.echo(`Algoliasearch-Helper release CANARY version`);
+
+  checkCleanWorkdir();
+  updateCanaryVersion(canaryVersion => {
     build();
-    updateChangelog(shell);
-    commitNewFiles(version);
-    publish();
-    goBackToDevelop();
+    publishOnNpm('canary');
+    revertStableVersion(packageJson.version, () => {
```

Comment:
Is this function completely resetting the state it was in before creating the canary release? (`version` in `package.json`, git staging, etc.)

It's quite hard to follow the release flow.

Reply:
Yes, it does. I agree that it's not the optimal solution. A better approach would be to build the complete package into a dedicated folder that we can discard at the end. Unfortunately, that would completely change the current release process. I tried to minimize the changes, hence the update + revert of the changes.

Reply:
here it's still the old version, since you require before you change things right?

Reply:
Yes it is.